Example/LiquidSwipe/LiquidSwipe.tsx

```
@@ -1,61 +1,88 @@
-      }
-      
-      // snapPoint
-      velocityX = (-velocityX)/(isBack.value * maxDist.value + (1-isBack.value) * 0.4 * maxDist.value);
-      const point = gestureProgress.value + 0.2 * velocityX;
-      snapPoint.set(Reanimated.clamp(point,[0, 1]));
-      // centerY
-      centerY.set(Reanimated.withWorklet(spring.worklet, [{}, {toValue: y}])); 
-    }, [gestureProgress, isBack, state, maxDist, snapPoint, spring, centerY]
+  const handler = useAnimatedGestureHandler(
+    {
+      onStart: (event, inputs, ctx) => {
+        'worklet';
+        // stop animating progress, this will also place "isBack" value in the
+        // final state (we update isBack in progress animation callback)
+        Reanimated.cancelAnimation(inputs.progress);
+        ctx.dragX = 0;
+        ctx.startY = inputs.isBack.value ? event.y : inputs.centerY.value;
+      },
+      onActive: (event, inputs, ctx) => {
+        'worklet';
+        const { centerY, isBack, progress } = inputs;
+        centerY.value = ctx.startY + event.translationY;
+        if (isBack.value) {
+          progress.value = Reanimated.interpolate(
+            event.translationX,
+            [0, inputs.maxDist.value],
+            [1, 0],
+            Extrapolate.CLAMP
+          );
+        } else {
+          progress.value = Reanimated.interpolate(
+            event.translationX,
+            [-inputs.maxDist.value, 0],
+            [0.4, 0],
+            Extrapolate.CLAMP
+          );
+        }
+      },
+      onEnd: (event, inputs, ctx) => {
+        'worklet';
+        const { initialWaveCenter, isBack, centerY, progress } = inputs;
+        let goBack;
+        if (isBack.value) {
+          goBack = progress.value > 0.5 ? 1 : 0;
+        } else {
+          // TODO: want to use a boolean here
+          goBack = progress.value > 0.2 ? 1 : 0;
+        }
+        centerY.value = Reanimated.withSpring(initialWaveCenter.value);
```

I think either both `initialWaveCenter` and `initialWaveCenter.value` should be valid here or withSpring should throw an error explaining user should use `.value` Thing. Because it's going to be a common mistake (I already have got an error because of that)